crates/oro-client/src/credentials.rs

```diff
@@ -0,0 +1,49 @@
+use std::{collections::HashMap, fmt::Debug};
+
+use crate::OroClientError;
+
+/**
+ * HTTP basic credentials (i.e. username & password)
+ */
+#[derive(Clone)]
+pub struct BasicAuthCredentials {
+    pub username: String,
+    pub password: String,
+}
+
+impl Debug for BasicAuthCredentials {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("username: {}, password: ***", self.username))
+    }
+}
+
+/**
+ * Different credential types supported by orogene
+ */
+#[derive(Clone, Debug)]
+pub enum Credentials {
+    Basic(BasicAuthCredentials),
```

Comment:
why not just `Basic { username: String, password: String }`?

Reply:
Because I forgot that this was possible. Need sleep or something.